internal/secret/secret.go

```diff
@@ -47,3 +49,25 @@ func Decode(cipherText string, decryptPassword bool) (plainText string) {
 
 	return
 }
+
+// DecodeAsUtf16 takes a cipher text and a boolean indicating whether to decrypt
+// and returns the resulting plain text as a byte array in UTF-16LE format which
+// is required when passing the secret to applications written using managed
+// code (C#), such as Azure Data Studio.
+func DecodeAsUtf16(cipherText string, decryptPassword bool) []byte {
+	var buf bytes.Buffer
+	secret := Decode(cipherText, decryptPassword)
+	runes := utf16.Encode([]rune(secret))
+
+	const BOM = '\ufffe' //LE BOM
+	var b [2]byte
+	b[0] = BOM >> 8
+	b[1] = BOM & 255
+	for _, r := range runes {
+		b[1] = byte(r >> 8)
+		b[0] = byte(r & 255)
+		buf.Write(b[:])
```

Comment:
>buf.Write(b[:]) [](http://example.com/codeflow?start=2&length=15)

Isn't this skipping a `Write` of the BOM?

Reply:
Yes, that's why it works!  Removed creation of the BOM, ADS doesn't want it. (auth fails if i write it here)